share/functions/__fish_cache_sourced_completions.fish

```diff
@@ -0,0 +1,45 @@
+function __fish_cache_sourced_completions
+    # Allow a `--name=foo` option which ends up in the filename.
+    argparse -s name= -- $argv
+    or return
```

Comment:
`name` is unused; I guess we don't need it at this point.

Maybe apply `path basename` before adding it to the cache path so this function never accidentally overwrites files outside the cache path.

If we want a full cache key we could serialize the entire argv but I guess that's not necessary.

Reply:
I mainly wanted to add an argparse here so it's easier to add options later.

I guess I can remove the name, but that would be the sort of thing we would add.